### DISCUSSION TITLE
allow sshuttle to run without chdiring into home directory

### DISCUSSION THREAD

**Discussion by estellnb** [sshuttle/sshuttle#605] [General]
There are some free ssh services out there which do not allow to chdir into the home directory. Nonetheless they can run as a socks port forwarder:

> ssh ssh-free.com-estellnb@us1.ssh-free.com -f -N -D1234

see: www.ssh-free.com

Why not make sshuttle work with such services?

**Comment by brianmay** [comment]
openssh doesn't support a socks backend.

I would actually like to see a version of sshuttle that can talk to a socks5 backend. For various reasons, e.g. reduces/less evidence on server that you are using sshuttle, which might be important for some applications. No yucky hacks to transfer Python code to remove. Less Python code and more C used code which might(?) improve performance. Support non-ssh based socks server (does anybody actually do this anymore?). etc

Not sure if this would be a replacement for the existing support or in addition to it. It depends if there is anything the ssh backend can do that a socks backend can't do.

I believe socks5 supports UDP and IPv6, which somewhat surprises me. 

As much as I would like to develop this, unfortunately I have no time to do this. But I see no technical reason why it couldn't be implemented.

**Comment by estellnb** [comment]
No, I do not mean to make sshuttle work with socks5. I mean to make sshuttle work with an ssh account where you can not log in and chdir to the user´s home directory. The above command which installs socks forwarding (-D1234) uses (-N) to avoid a chdir into the home directory. You can use the (-N) switch independently from the socks siwtch -D1234. (-f) does finally run ssh in the background.

**Comment by brianmay** [comment]
Look up  the ssh documentation. `-N` means:

> Do not execute a remote command.  This is useful for just forwarding ports.

This will not work with sshuttle, which currently needs to run Python code on the remote host. socks5 support would solve this.
